File: 1281-subtract-the-product-and-sum-of-digits-of-an-integer/1281-subtract-the-product-and-sum-of-digits-of-an-integer.py
class Solution:
    def subtractProductAndSum(self, n: int) -> int:
        product = 1
        sum = 0
        i = 0
        
        while (n != 0):
            i = n % 10
            product *= i
            sum += i
            
            n //= 10
            
        return product - sum

File: 1281-subtract-the-product-and-sum-of-digits-of-an-integer/1281-subtract-the-product-and-sum-of-digits-of-an-integer.java
class Solution {
    public int subtractProductAndSum(int n) {
        int product = 1, sum = 0;
        int i;
        
        while(n != 0) {
            //use modulo to get the remainder
            i = n % 10;
            product *= i;
            sum += i;
            //divide by 10 to take advantage of truncation
            n /= 10; 
        }
        return product - sum;
    }
}